frame/support/src/traits/metadata.rs

```diff
@@ -125,47 +115,75 @@ impl PalletVersion {
+		let key = Self::storage_key::<P>();
 
 		crate::storage::unhashed::put(&key, self);
 	}
+
+	/// Get the storage version of the given pallet from the storage.
+	///
+	/// It will use the storage key that is associated with the given `Pallet`.
+	///
+	/// # Panics
+	///
+	/// This function will panic iff `Pallet` can not be found by `PalletInfo`.
+	/// In a runtime that is put together using
+	/// [`construct_runtime!`](crate::construct_runtime) this should never happen.
+	///
+	/// It will also panic if this function isn't executed in an externalities
+	/// provided environment.
+	pub fn get<P: PalletInfoAccess>() -> Self {
+		let key = Self::storage_key::<P>();
+
+		crate::storage::unhashed::get_or_default(&key)
+	}
 }
 
-impl sp_std::cmp::PartialOrd for PalletVersion {
-	fn partial_cmp(&self, other: &Self) -> Option<sp_std::cmp::Ordering> {
-		let res = self
-			.major
-			.cmp(&other.major)
-			.then_with(|| self.minor.cmp(&other.minor).then_with(|| self.patch.cmp(&other.patch)));
+impl PartialEq<u16> for StorageVersion {
+	fn eq(&self, other: &u16) -> bool {
+		self.0 == *other
+	}
+}
 
-		Some(res)
+impl PartialOrd<u16> for StorageVersion {
+	fn partial_cmp(&self, other: &u16) -> Option<sp_std::cmp::Ordering> {
+		Some(self.0.cmp(other))
 	}
 }
 
-/// Provides version information about a pallet.
+/// Provides information about the storage version of a pallet.
 ///
-/// This trait provides two functions for returning the version of a
-/// pallet. There is a state where both functions can return distinct versions.
-/// See [`GetPalletVersion::storage_version`] for more information about this.
-pub trait GetPalletVersion {
-	/// Returns the current version of the pallet.
-	fn current_version() -> PalletVersion;
-
-	/// Returns the version of the pallet that is stored in storage.
-	///
-	/// Most of the time this will return the exact same version as
-	/// [`GetPalletVersion::current_version`]. Only when being in
-	/// a state after a runtime upgrade happened and the pallet did
-	/// not yet updated its version in storage, this will return a
-	/// different(the previous, seen from the time of calling) version.
-	///
-	/// See [`PalletVersion`] for more information.
-	///
-	/// # Note
-	///
-	/// If there was no previous version of the pallet stored in the state,
-	/// this function returns `None`.
-	fn storage_version() -> Option<PalletVersion>;
+/// It differentiates between current and active storage version. Both should be only out of sync
+/// when a new runtime upgrade was applied and the runtime migrations did not yet executed.
+/// Otherwise it means that the pallet works with an unsupported storage version and unforeseen
+/// stuff can happen.
+///
+/// The current storage version is the version of the pallet as supported at runtime. The active
+/// storage version is the version of the pallet in the storage.
+///
+/// It is required to update the active storage version manually when a migration was applied.
```

Comment:
using the same name as functions makes it more easy to grasp maybe
```suggestion
/// It differentiates between current and on-chain storage version. Both should be only out of sync
/// when a new runtime upgrade was applied and the runtime migrations did not yet executed.
/// Otherwise it means that the pallet works with an unsupported storage version and unforeseen
/// stuff can happen.
///
/// The current storage version is the version of the pallet as supported at runtime. The active
/// storage version is the version of the pallet in the storage.
///
/// It is required to update the on-chain storage version manually when a migration was applied.
```

Reply:
we could also link to `StorageVersion::put` for manually inserting the version.